### PR TITLE
Added missing description for Translation types.

### PR DESCRIPTION
**Pull request by Smit-Parmar**
I want to merge this change because of adding missing descriptions to translation types.
Part of #13142 

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/graphql/translations/types.py [hunk 1]
```diff
@@ -116,8 +126,13 @@ def resolve_attribute(root: attribute_models.Attribute, _info):
...
     ModelObjectType[attribute_models.AttributeValue]
 ):
-    id = graphene.GlobalID(required=True)
-    name = graphene.String(required=True)
+    id = graphene.GlobalID(
+        required=True, description="The ID of the attribute value translatable content."
+    )
+    name = graphene.String(
+        required=True,
+        description="The name of the attribute value translatable content.",
```
Comment (maarcingebala) [inline saleor/graphql/translations/types.py:134]: Name of the attribute to translate.

File: saleor/graphql/translations/types.py [hunk 2]
```diff
@@ -151,17 +166,26 @@ def resolve_attribute(root: attribute_models.AttributeValue, info):
 class ProductVariantTranslation(
     BaseTranslationType[product_models.ProductVariantTranslation]
 ):
-    id = graphene.GlobalID(required=True)
-    name = graphene.String(required=True)
+    id = graphene.GlobalID(
+        required=True, description="The ID of the product variant translation."
+    )
+    name = graphene.String(
+        required=True, description="The name of the product variant translation."
```
Comment (maarcingebala) [inline saleor/graphql/translations/types.py:173]: Translated product variant name.

File: saleor/graphql/translations/types.py [hunk 3]
```diff
@@ -151,17 +166,26 @@ def resolve_attribute(root: attribute_models.AttributeValue, info):
...
 
 class ProductVariantTranslatableContent(ModelObjectType[product_models.ProductVariant]):
-    id = graphene.GlobalID(required=True)
-    name = graphene.String(required=True)
+    id = graphene.GlobalID(
+        required=True, description="The ID of the product variant translatable content."
+    )
+    name = graphene.String(
+        required=True,
+        description="The name of the product variant translatable content.",
```
Comment (maarcingebala) [inline saleor/graphql/translations/types.py:187]: Name of the product variant to translate.

File: saleor/graphql/translations/types.py [hunk 4]
```diff
@@ -198,10 +222,14 @@ def resolve_attribute_values(root: product_models.ProductVariant, info):
...
 
 class ProductTranslation(BaseTranslationType[product_models.ProductTranslation]):
-    id = graphene.GlobalID(required=True)
-    seo_title = graphene.String()
-    seo_description = graphene.String()
-    name = graphene.String()
+    id = graphene.GlobalID(
+        required=True, description="The ID of the product translation."
+    )
+    seo_title = graphene.String(description="SEO title of the product.")
```
Comment (maarcingebala) [inline saleor/graphql/translations/types.py:228]: Translated SEO title.

File: saleor/graphql/translations/types.py [hunk 5]
```diff
@@ -198,10 +222,14 @@ def resolve_attribute_values(root: product_models.ProductVariant, info):
...
+    seo_description = graphene.String(
+        description="SEO description of the product translation."
```
Comment (maarcingebala) [inline saleor/graphql/translations/types.py:230]: Translated SEO description.

File: saleor/graphql/translations/types.py [hunk 6]
```diff
@@ -198,10 +222,14 @@ def resolve_attribute_values(root: product_models.ProductVariant, info):
...
+    )
+    name = graphene.String(description="Name of the product translation.")
```
Comment (maarcingebala) [inline saleor/graphql/translations/types.py:232]: Translated product name.

File: saleor/graphql/translations/types.py [hunk 7]
```diff
@@ -223,10 +251,18 @@ def resolve_description_json(root: product_models.ProductTranslation, _info):
...
 class ProductTranslatableContent(ModelObjectType[product_models.Product]):
-    id = graphene.GlobalID(required=True)
-    seo_title = graphene.String()
-    seo_description = graphene.String()
-    name = graphene.String(required=True)
+    id = graphene.GlobalID(
+        required=True, description="The ID of the product translatable content."
+    )
+    seo_title = graphene.String(
+        description="SEO title of the product translatable content."
```
Comment (maarcingebala) [inline saleor/graphql/translations/types.py:258]: SEO title to translate.

File: saleor/graphql/translations/types.py [hunk 8]
```diff
@@ -223,10 +251,18 @@ def resolve_description_json(root: product_models.ProductTranslation, _info):
...
+    )
+    seo_description = graphene.String(
+        description="SEO description of the product translatable content."
```
Comment (maarcingebala) [inline saleor/graphql/translations/types.py:261]: SEO description to translate.

File: saleor/graphql/translations/types.py [hunk 9]
```diff
@@ -223,10 +251,18 @@ def resolve_description_json(root: product_models.ProductTranslation, _info):
...
+    )
+    name = graphene.String(
+        required=True, description="Name of the product translatable content."
```
Comment (maarcingebala) [inline saleor/graphql/translations/types.py:264]: Product's name to translate.

**Review by maarcingebala — COMMENTED** [review]
Thanks for opening the PR! I added couple comments to make the descriptions more precise, but the rest of the changes here could be unified similarly.
- ObjectTranslation types are translated properties of the object, so the descriptions should follow the format: "Translated <field> of <type>" e.g. "Translated name of the product"
- ObjectTranslatableContent types represent properties to be translated, so the format can be "<Field> to translate" e.g. "Product name to translate".

**Comment by Smit-Parmar** [comment]
Thanks @maarcingebala . Updated the changes.

File: saleor/graphql/translations/types.py [hunk 10]
```diff
@@ -271,10 +301,12 @@ def resolve_attribute_values(root: product_models.Product, info):
...
-    id = graphene.GlobalID(required=True)
-    seo_title = graphene.String()
-    seo_description = graphene.String()
-    name = graphene.String()
+    id = graphene.GlobalID(
+        required=True, description="The ID of the collection translation."
+    )
+    seo_title = graphene.String(description="Translated SEO title.")
+    seo_description = graphene.String(description="Translated SEO description.")
+    name = graphene.String(description="Translated colletion name.")
```
Comment (maarcingebala) [inline saleor/graphql/translations/types.py:309]: Typo:
```suggestion
    name = graphene.String(description="Translated collection name.")
```

File: saleor/graphql/translations/types.py [hunk 11]
```diff
@@ -539,27 +589,35 @@ def resolve_sale(root: discount_models.Sale, _info):
...
         interfaces = [graphene.relay.Node]
 
 
 class MenuItemTranslatableContent(ModelObjectType[menu_models.MenuItem]):
-    id = graphene.GlobalID(required=True)
-    name = graphene.String(required=True)
+    id = graphene.GlobalID(
+        required=True, description="The ID of the menu item translatable content."
+    )
+    name = graphene.String(required=True, description="Name of the menu to translate.")
```
Comment (maarcingebala) [inline saleor/graphql/translations/types.py:620]: Name of the menu item to translate.

**Review by maarcingebala — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Pull request merged**